openmc/mgxs_library.py

```diff
@@ -553,23 +585,526 @@ def nu_fission(self, nu_fission):
+                raise ValueError(msg)
+            # Get openmc.mgxs.get_xs() arguments from kwargs
+            # nuclides, xs_type, and value will have sane defaults but can be
+            # overridden by kwards
+            if 'nuclides' in kwargs:
+                nuclides = kwargs['nuclides']
+            else:
+                nuclides = 'sum'
+            if 'xs_type' in kwargs:
+                xs_type = kwargs['xs_type']
+            else:
+                xs_type = 'macro'
+            if 'value' in kwargs:
+                value = kwargs['value']
+            else:
+                value = 'mean'
+            # subdomains is required from the kwargs as this is specific to
+            # this XSdata object.
+            if 'subdomains' in kwargs:
+                subdomains = kwargs['subdomains']
+            else:
+                msg = "Argument 'subdomains' is required"
+                raise ValueError(msg)
+
+            if self._representation is 'isotropic':
+                self._nu_fission = nu_fission.get_xs(subdomains=subdomains,
+                                                     nuclides=nuclides,
+                                                     xs_type=xs_type,
+                                                     value=value)
+            elif self._representation is 'angle':
+                # Not yet implemented as MGXS do not yet support this
+                pass
+
+            self._use_chi = True
+
+        else:
+            # nu_fission can be given as a vector or a matrix
+            # Vector is used when chi also exists.
+            # Matrix is used when chi does not exist.
+            # We have to check that the correct form is given, but only if
+            # chi already has been set.  If not, we just check that this is OK
+            # and set the use_chi flag accordingly
+
+            # First lets set our dimensions here since they get used repeatedly
+            # throughout this code.
+            if self._representation is 'isotropic':
+                shape_vec = (self._energy_groups.num_groups,)
+                shape_mat = (self._energy_groups.num_groups,
+                             self._energy_groups.num_groups)
+            elif self._representation is 'angle':
+                shape_vec = (self._num_polar, self._num_azimuthal,
+                             self._energy_groups.num_groups)
+                shape_mat = (self._num_polar, self._num_azimuthal,
+                             self._energy_groups.num_groups,
+                             self._energy_groups.num_groups)
+
+            # Begin by checking the case when chi has already been given and
+            # thus the rules for filling in nu_fission are set.
+            if self._use_chi is not None:
+                if self._use_chi:
+                    shape = shape_vec
+                else:
+                    shape = shape_mat
+                if nu_fission.shape != shape:
```

Comment:
Replace this with an expression using the formal type-checking interface in `openmc.checkvalue`